scripts/release.ts

```diff
@@ -93,71 +105,156 @@ function init(args) {
+  run(commitCmd);
 
-  rmSync(temporaryDir, { recursive: true });
+  return version;
+}
+
+function makePullBody(diff: string) {
+  const bodyFile = fileURLToPath(
+    new URL("release-pull-description.md", import.meta.url)
+  );
+  const body = [
+    readFileSync(bodyFile, { encoding: "utf-8" }),
+    "```diff",
+    diff,
+    "```",
+  ].join("\n");
+  return body;
 }
 
 function update(args) {
-  // TODO: Rebase
-  // TODO: Run `npm run build`
-  // TODO: Commit results of `npm run build`
-  // TODO: Update PR diff
-  throw Error("Not implemented");
+  preflight({ expectedPull: args.pr, targetRepo: args.targetRepo });
+  build();
+
+  logger.verbose("Adding rebase-in-progress notice to PR description");
+  const { body } = JSON.parse(
+    execSync(
+      `gh pr view --repo="${args.targetRepo}" "${args.pr}" --json body`,
+      {
+        encoding: "utf-8",
+      }
+    )
+  );
+  const notice = "⛔️ Update in progress! ⛔️\n";
+  const editBodyCmd = `gh pr edit --repo="${args.targetRepo}" "${args.pr}" --body-file=-`;
+  execSync(editBodyCmd, {
+    input: [notice, body].join("\n\n"),
+    stdio: ["pipe", "inherit", "inherit"],
+  });
+
+  logger.verbose("Rebasing");
+  try {
+    run(`git rebase ${args.base}`);
+  } catch (err) {
+    logger.error("Rebasing failed. Abandoning PR.");
+    run(`git rebase --abort`);
+    run(
+      `gh pr comment --repo="${args.targetRepo}" "${args.pr}" --body="😱 Rebasing failed. Closing this PR. 😱"`
+    );
+    run(`gh pr close --repo="${args.targetRepo}" "${args.pr}"`);
+    process.exit(1);
+  }
+
+  const diff = diffJson();
+
+  if (args.bump) {
+    const newVersion = bumpVersion(args.bump);
+
+    logger.info("Pushing release branch");
+    run(`git push origin HEAD`);
+
+    logger.verbose("Updating PR title");
+    run(
+      `gh pr edit --repo="${args.targetRepo}" "${args.pr}" --title="${pullTitleBase}${newVersion}"`
+    );
+  }
+
+  logger.verbose("Removing update-in-progress notice from PR description");
+  execSync(editBodyCmd, { input: body, stdio: ["pipe", "inherit", "inherit"] });
+
+  const updatedBody = makePullBody(diff);
+  execSync(
+    `gh pr edit --repo="${args.targetRepo}" "${args.pr}" --body-file=-`,
+    {
+      input: updatedBody,
+      stdio: ["pipe", "inherit", "inherit"],
+    }
+  );
 }
 
 function publish(args) {
-  // TODO: Run `npm publish …`
-  throw Error("Not implemented");
+  preflight({ expectedBranch: "main" });
+  build();
+  const { version } = readPackageJSON(packages["web-features"]);
+  const tag = `web-features/${version}`;
+  run(`git tag --annotate "${tag}" --message="web-features ${version}"`);
+  run(`git push origin ${tag}`);
+
+  logger.info("Publishing release");
+  let publishCmd = `npm publish`;
+  if (args.dryRun) {
+    publishCmd = `${publishCmd} --dry-run`;
+  }
+  execSync(publishCmd, { cwd: packages["web-features"], stdio: "inherit" });
 }
 
 function run(cmd: string) {
   execSync(cmd, { stdio: "inherit" });
 }
```

Comment:
This file isn't very well organized—the functions are in no particular order. But it made the diff even harder to follow. Perhaps something to tidy up either just before merging, or in a separate PR.